src/main/java/io/stargate/sgv2/jsonapi/service/operation/model/impl/ReadDocument.java

```diff
@@ -10,5 +12,21 @@
  * @param id Document Id identifying the document
  * @param txnId Unique UUID resenting point in time of a document, used for LWT transactions
  * @param document JsonNode representation of the document
+ * @param sortColumns List<JsonNode> Serialized sort column value
+ * @param docJsonValue Grpc column value for doc_json.
  */
-public record ReadDocument(DocumentId id, UUID txnId, JsonNode document) {}
+public record ReadDocument(
+    DocumentId id,
+    UUID txnId,
+    JsonNode document,
+    List<JsonNode> sortColumns,
+    QueryOuterClass.Value docJsonValue) {
+  public ReadDocument(DocumentId id, UUID txnId, JsonNode document) {
```

Comment:
Instead of separate constructors, maybe create 2 factory methods so name can describe kind of document created -- since we are now creating 2 kinds of `ReadDocument`s? They can both use one canonical constructor.

Reply:
Will make the change